recipes/server_extend.rb

```diff
@@ -16,30 +16,32 @@
       Chef::Log.warn("Unable to find a chef node for #{peer}")
       next
     end
-    peer_bricks = chef_node['gluster']['server']['bricks'].select { |brick| brick.include? volume_name }
+    peer_bricks = chef_node['gluster']['server'][volume_name]['bricks'].select { |brick| brick.include? volume_name }
     brick_count += (peer_bricks.count || 0)
     peer_bricks.each do |brick|
       Chef::Log.info("Checking #{peer}:#{brick}")
       unless brick_in_volume?(peer, brick, volume_name)
-        node.default['gluster']['server']['bricks_waiting_to_join'] << " #{peer}:#{brick}"
+        node.default['gluster']['server']['volumes'][volume_name]['bricks_waiting_to_join'] << " #{peer}:#{brick}"
```

Comment:
think this might relate to `node['gluster']['server']['volumes'][volume_name]['bricks_waiting_to_join']` not being defined as i mentioned in another comment